lib/takeaway.rb

```diff
@@ -0,0 +1,52 @@
+require './lib/restaurant'
+
+class TakeAway
+  attr_reader :restaurant, :dish, :basket
+
+  def initialize (restaurant = Restaurant.new)
+    @restaurant = restaurant
+    @dish = ""
+    @basket = Hash.new(0)
+  end
+
+  def read_menu
+    restaurant.read_menu
+  end
+
+  def order(dish, qty=1)
+    raise "This dish is not available on menu" if not_on_menu?(dish)
```

Comment:
Style/SignalException: Use `fail` instead of `raise` to signal exceptions. (https://github.com/bbatsov/ruby-style-guide#fail-method)